docs/sdk/src/reference_docs/chain_spec_genesis.rs

```diff
@@ -1,4 +1,110 @@
+#![doc = docify::embed!("./src/reference_docs/chain_spec_runtime/src/pallets.rs", pallet_bar_build)]
+
+//! GenesisConfig may contain also more complicated types including nested structs or enums, as in
+//! the example for `pallet_foo`:
+#![doc = docify::embed!("./src/reference_docs/chain_spec_runtime/src/pallets.rs", pallet_foo_GenesisConfig)]
+
+//! Please note that fields of `GenesisConfig` may not be directly mapped to storage items:
+#![doc = docify::embed!("./src/reference_docs/chain_spec_runtime/src/pallets.rs", pallet_foo_build)]
+
+//! ## `GenesisConfig` for `runtimes`
+
+//! The runtime genesis config struct consists of configs for every pallet. For [_demonstration
+//! runtime_][`chain_spec_guide_runtime`] used in this guide, it consists of `SystemConfig`,
+//! `BarConfig` and `FooConfig`. This structure was automatically generated by macro and it can be
+//! sneaked peaked here: [`RuntimeGenesisConfig`].
+//!
+//! Runtime exposes a dedicated runtime API for interacting with its genesis config:
+//! [`sp_genesis_builder::GenesisBuilder`]. Implementation shall be provided within
+//! [`sp_api::impl_runtime_apis`] macro, typically making use of some helpers provided:
+//! [`build_state`], [`get_preset`].
+//! Usual implementation of [`sp_genesis_builder::GenesisBuilder`] is as follows:
+#![doc = docify::embed!("./src/reference_docs/chain_spec_runtime/src/runtime.rs", runtime_impl)]
+
+//! Please note that two functions are customized `preset_names` and `get_preset`. The first one
+//! just provdes a vec of names of supported presets, while the latter one delegats the call to
+//! function that maps the name to preset: [`chain_spec_guide_runtime::presets::get_builtin_preset`]
+#![doc = docify::embed!("./src/reference_docs/chain_spec_runtime/src/presets.rs", get_builtin_preset)]
+
+//! ## Genesis state presets for `runtime`
+
+//! Rutime may provide many flavors of initial genesis state. This may be useful for predefined
+//! testing networks, local development, or CI integration tests. Predefined genesis state may
+//! contain a list of pre-funded accounts, predefined authorithies for consensue, sudo key and many
+//! others useful for testing.
+
+//! Internally presets can be provided in a number of ways:
+//! - json in string form:
+#![doc = docify::embed!("./src/reference_docs/chain_spec_runtime/src/presets.rs", preset_1)]
+//! - json using runtime types to serialize values:
+#![doc = docify::embed!("./src/reference_docs/chain_spec_runtime/src/presets.rs", preset_2)]
+#![doc = docify::embed!("./src/reference_docs/chain_spec_runtime/src/presets.rs", preset_3)]
+//! It is worth noting that preset does not have to be the full `RuntimeGenesisConfig`, in that
+//! sense that it does not have to contain all the keys of the struct. The preset is acutally a json
+//! patch that will be merged with default value of `RuntimeGenesisConfig`. This approach should
```

Comment:
to be nit picky I would say the concept of "Default" for genesis config is not explained up until this point. 

Worth saying:

- `GenesisConfig` of all pallets _must_ have a `Defualt`
- which is aggregated into `RuntimeGenesisConfig`'s `Default`.
- this defined what the `get_preset` runtime api would return with `id: None`

Reply:
good point, thx

Reply:
added _default_ section in: c5d1f865eb4